core/src/main/java/org/elasticsearch/common/SwallowsExceptions.java

```diff
@@ -0,0 +1,31 @@
+package org.elasticsearch.common;
+
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Target;
+
+/**
+ * Annotation to suppress broken catch-block swallowing for a method or class
+ */
+@Target({ElementType.CONSTRUCTOR, ElementType.METHOD, ElementType.TYPE})
```

Comment:
you can. Its the same situation as forbidden apis.

Originally i developed this to use type annotations, e.g. `catch(@Swallowed Exception e) { ... `, but those have numerous problems:
* you can type-annotate an exception parameter in a clinit, it simply never makes it into the bytecode at all
* in the case of unreachable code, the annotation is also lost
* for finally blocks and try-with-resources statements, your code is duplicated, but not the annotation.

So basically, those are so unreliable, that we'd have to be lenient, and couldnt fail if the annotation was unnecessary. I think thats really important :)